quickwit-config/src/index_config.rs

```diff
@@ -0,0 +1,228 @@
+// Copyright (C) 2021 Quickwit, Inc.
+//
+// Quickwit is offered under the AGPL v3.0 and as commercial software.
+// For commercial licensing, contact us at hello@quickwit.io.
+//
+// AGPL:
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+use std::ffi::OsStr;
+use std::path::Path;
+
+use anyhow::{bail, Context};
+use byte_unit::Byte;
+use quickwit_index_config::FieldMappingEntry;
+use serde::{Deserialize, Serialize};
+
+#[derive(Debug, Serialize, Deserialize)]
+pub struct DocMapping {
+    pub field_mappings: Vec<FieldMappingEntry>,
+    pub tag_fields: Vec<String>,
+    pub store_source: bool,
+}
+
+#[derive(Debug, Serialize, Deserialize, PartialEq)]
+pub struct IndexingResources {
+    pub num_threads: usize,
+    pub heap_size: Byte,
+}
+
+#[derive(Debug, Serialize, Deserialize, PartialEq)]
+pub struct MergePolicy {
+    pub demux_factor: usize,
+    pub merge_factor: usize,
+    pub max_merge_factor: usize,
+    pub min_level_num_docs: usize,
+}
+
+#[derive(Debug, Serialize, Deserialize, PartialEq)]
+pub struct IndexingSettings {
+    pub demux_field: String,
+    pub timestamp_field: String,
+    pub commit_timeout_secs: usize,
+    /// The maximum number of documents allowed in a split.
+    pub split_max_num_docs: usize,
+    pub merge_policy: MergePolicy,
+    pub resources: IndexingResources,
+}
+
+#[derive(Debug, Serialize, Deserialize, PartialEq)]
+pub struct SearchSettings {
+    pub default_search_fields: Vec<String>,
+}
+
+#[derive(Serialize, Deserialize)]
+pub struct SourceConfig {
+    pub source_id: String,
+    pub source_type: String,
+    pub params: toml::Value,
```

Comment:
what are the advantages of using `toml::Value` over a json value?

Reply:
Limited, primarily support for date and time types. I thought toml would be a first-class citizen... maybe not :)